include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_kernels.h

```diff
@@ -0,0 +1,722 @@
+        __p_global_hist(__p_global_hist), __p_group_hists(__p_group_hists),
+        __in_pack(__in_pack), __out_pack(__out_pack) {}
+
+    template<typename _SimdPack>
+    inline auto
+    __load_simd_pack(_SimdPack& __pack, ::std::uint32_t __wg_id, ::std::uint32_t __wg_size, ::std::uint32_t __lid) const
+    {
+        const _GlobOffsetT __offset = __data_per_work_item * (__wg_id * __wg_size + __lid);
+        __load_simd</*__sort_identity_residual=*/true>(__pack.__keys, __rng_data(__in_pack.__keys_rng()), __offset);
+        if constexpr (__has_values)
+        {
+            __load_simd</*__sort_identity_residual=*/false>(__pack.__vals, __rng_data(__in_pack.__vals_rng()), __offset);
+        }
+    }
+
+    template<bool __sort_identity_residual, typename _T, typename _InSeq>
+    inline void
+    __load_simd(_SimdT<_T, __data_per_work_item>& __simd, const _InSeq& __in_seq, _GlobOffsetT __glob_offset) const
+    {
+        static_assert(__data_per_work_item % __data_per_step == 0);
+
+        __dpl_esimd::__ns::simd<::std::uint16_t, __data_per_step> __lane_id(0, 1);
+        bool __is_full_block = (__glob_offset + __data_per_work_item) < __n;
+        if (__is_full_block)
+        {
+            _ONEDPL_PRAGMA_UNROLL
+            for (::std::uint32_t __s = 0; __s < __data_per_work_item; __s += __data_per_step)
+            {
+                __dpl_esimd::__ns::simd<::std::uint32_t, __data_per_step> __offset = __glob_offset + __s + __lane_id;
+                __simd.template select<__data_per_step, 1>(__s) =
+                    __dpl_esimd::__gather<_T, __data_per_step>(__in_seq, __offset, 0);
+            }
+        }
+        else
+        {
+            _ONEDPL_PRAGMA_UNROLL
+            for (::std::uint32_t __s = 0; __s < __data_per_work_item; __s += __data_per_step)
+            {
+                __dpl_esimd::__ns::simd<::std::uint32_t, __data_per_step> __offset = __glob_offset + __s + __lane_id;
+                __dpl_esimd::__ns::simd_mask<__data_per_step> __m = __offset < __n;
+                auto __gathered = __dpl_esimd::__gather<_T, __data_per_step>(__in_seq, __offset, 0, __m);
+                if constexpr (__sort_identity_residual)
+                {
+                    constexpr _T __default_item = __sort_identity<_T, __is_ascending>();
+                    auto __default = __dpl_esimd::__ns::simd<_T, __data_per_step>(__default_item);
+                    __simd.template select<__data_per_step, 1>(__s) =__dpl_esimd::__ns::merge(__gathered, __default, __m);
+                }
+                else
+                {
+                    __simd.template select<__data_per_step, 1>(__s) = __gathered;
+                }
+            }
+        }
+    }
+
+    static inline __dpl_esimd::__ns::simd<::std::uint32_t, 32>
+    __match_bins(const __dpl_esimd::__ns::simd<::std::uint32_t, 32>& __bins, ::std::uint32_t __local_tid)
+    {
+        __dpl_esimd::__ns::fence<__dpl_esimd::__ns::fence_mask::sw_barrier>();
```

Comment:
It would be good to add comments to explain the usage of `__dpl_esimd::__ns::fence` here and the other places it is present.

Reply:
I have added comments